tap_woo/streams.py

```diff
@@ -591,29 +330,45 @@ class SubscriptionsStream(wooStream):
         th.Property("end_date_gmt", th.DateTimeType),
         th.Property("resubscribed_from", th.StringType),
         th.Property("resubscribed_subscription", th.StringType),
-        th.Property("removed_line_items", th.ArrayType(th.IntegerType))
+        th.Property("removed_line_items", th.ArrayType(th.IntegerType)),
     ).to_dict()
 
+    def post_process(self, row: dict, context: dict | None = None) -> dict | None:
+        list_date_fields = [
+            'trial_end_date_gmt',
+            'next_payment_date_gmt',
+            'last_payment_date_gmt',
+            'cancelled_date_gmt',
+            'end_date_gmt'
+        ]
+        for date_time_field in list_date_fields:
+            if date_time_field in row and not row[date_time_field]:
+                del row[date_time_field]
```

Comment:
It was failing when it was trying to convert the empty values `''` to datetime. 
```

FAILED tests/test_core.py::TestTapwoo::test_tap_stream_attribute_is_datetime[subscriptions.trial_end_date_gmt] - AssertionError: Unable to parse value ('') with datetime parser.
FAILED tests/test_core.py::TestTapwoo::test_tap_stream_attribute_is_datetime[subscriptions.next_payment_date_gmt] - AssertionError: Unable to parse value ('') with datetime parser.
FAILED tests/test_core.py::TestTapwoo::test_tap_stream_attribute_is_datetime[subscriptions.last_payment_date_gmt] - AssertionError: Unable to parse value ('') with datetime parser.
FAILED tests/test_core.py::TestTapwoo::test_tap_stream_attribute_is_datetime[subscriptions.cancelled_date_gmt] - AssertionError: Unable to parse value ('') with datetime parser.
FAILED tests/test_core.py::TestTapwoo::test_tap_stream_attribute_is_datetime[subscriptions.end_date_gmt] - AssertionError: Unable to parse value ('') with datetime parser.
```
We have two options, set everything in the field to string or delete the empty values and force it to be null. So I choose the second one removing the datetime empty rows to avoid the attempt to convert it.

> Should we do something more thorough, like look for any schema property of DateTimeType and apply this functionality, so we don't have to keep a hardcoded list of problem fields?

We are already defining the schema and some of the datetime fields can't be empty like the create datetime. Do you know more fields that could be empty? It's better to limit the list because we increase the processing time if we try to check all date fields.